examples/storage_usage/main.go

```diff
@@ -0,0 +1,171 @@
+/*
+ * Flow Go SDK
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go-sdk/templates"
+	"google.golang.org/grpc"
+	"strings"
+
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/client"
+	"github.com/onflow/flow-go-sdk/examples"
+)
+
+func main() {
+	StorageUsageDemo()
+}
+
+func StorageUsageDemo() {
+	ctx := context.Background()
+	flowClient, err := client.New("127.0.0.1:3569", grpc.WithInsecure())
+	examples.Handle(err)
+
+	serviceAcctAddr, serviceAcctKey, serviceSigner := examples.ServiceAccount(flowClient)
+
+	// A contract that defines a resource with a string so its easier to demonstrate adding resources of different lenght
+	contract := `
+		pub contract StorageDemo {
+			pub resource StorageTestResource {
+				pub let s: String
+				init(s: String) {
+					self.s = s
+				}
+			}
+			pub fun createStorageTestResource(_ s: String): @StorageTestResource {
+				return <- create StorageTestResource(s: s)
+			}
+		}
+	`
+	privateKey := examples.RandomPrivateKey()
+
+	key := flow.NewAccountKey().
+		SetPublicKey(privateKey.PublicKey()).
+		SetSigAlgo(privateKey.Algorithm()).
+		SetHashAlgo(crypto.SHA3_256).
+		SetWeight(flow.AccountKeyWeightThreshold)
+
+	keySigner := crypto.NewInMemorySigner(privateKey, key.HashAlgo)
+
+	demoAccount := examples.CreateAccountWithContracts(flowClient,
+		[]*flow.AccountKey{key}, []templates.Contract{{
+			Name:   "StorageDemo",
+			Source: contract,
+		}})
+	serviceAcctKey.SequenceNumber++
+
+	// try to save a very large resource to the demoAccount
+	txId := sendSaveLargeResourceTransaction(
+		ctx,
+		flowClient,
+		serviceAcctAddr,
+		serviceAcctKey,
+		serviceSigner,
+		demoAccount,
+		keySigner)
+
+	result := examples.WaitForSeal(ctx, flowClient, txId)
+
+	if result.Error == nil {
+		fmt.Println("Storage limits are off")
+		return
+	}
+
+	if !strings.HasPrefix(result.Error.Error(), "execution error code 11") {
```

Comment:
"error code 11" seems a bit like a magic number. Would it be possible to improve this check so it gets tries to get the code of the error, and if it is an error with a code, compares the code with a constant (that has a descriptive name)?

Reply:
So a function that looks for "code #" and returns the number if found. Should that be in the examples?

Reply:
Hmm, I meant on a type/field level, the function/statement shouldn't parse the error message/string. 

Where is code 11 defined / what does it mean?